Source/Scene/Scene.js

```diff
@@ -1196,19 +1327,34 @@ define([
             scene._sunBloom = false;
         }
 
+        // Manage celestial and terrestrial environment effects.
         var skyBoxCommand = (frameState.passes.render && defined(scene.skyBox)) ? scene.skyBox.update(context, frameState) : undefined;
         var skyAtmosphereCommand = (frameState.passes.render && defined(scene.skyAtmosphere)) ? scene.skyAtmosphere.update(context, frameState) : undefined;
         var sunCommand = (frameState.passes.render && defined(scene.sun)) ? scene.sun.update(scene) : undefined;
-        var sunVisible = isVisible(sunCommand, frameState);
 
+        // Preserve the reference to the original frame buffer.
         var originalFramebuffer = passState.framebuffer;
 
-        var clear = scene._clearColorCommand;
-        var clearDepth = scene._depthClearCommand;
+        // Create a working frustum from the original camera frustum.
+        var frustum;
+        if (defined(camera.frustum.fov)) {
+            frustum = camera.frustum.clone(scratchPerspectiveFrustum);
+        } else if (defined(camera.frustum.infiniteProjectionMatrix)){
+            frustum = camera.frustum.clone(scratchPerspectiveOffCenterFrustum);
+        } else {
+            frustum = camera.frustum.clone(scratchOrthographicFrustum);
+        }
 
-        Color.clone(clearColor, clear.color);
-        clear.execute(context, passState);
+        // Clear the pass state frame buffer.
+        var clearColorCommand = scene._clearColorCommand;
+        Color.clone(clearColor, clearColorCommand.color);
+        clearColorCommand.execute(context, passState);
 
+        // Update globe depth rendering based on the current context and clear the globe depth frame buffer.
+        scene._globeDepth.update(context);
+        scene._globeDepth.clear(context, passState, clearColor);
+
+        // Determine if there are any translucent surfaces in any of the frustum slices.
```

Comment:
I know you've introduced the term "slice" throughout, but I would be cautious in doing so.  We have been careful to be consistent about "frustum" vs "entire frustum", and introducing "slice" as another word for "frustum" is going to confuse people new to the code.